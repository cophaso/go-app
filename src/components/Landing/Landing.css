* {
  box-sizing: border-box;
}

body {
  font-family: "Open Sans",
  sans-serif;
  font-size: 20px;
  text-align: center;
  line-height: 1.5;
}

.header-landing {
  margin-top: -100px;
  padding: 100px 10px;
  background-color: #ffffff;
  background-image: url('../../images/LandingPage1.jpeg');
  background-size:100% 100%;
  background-repeat: no-repeat;
  background-position:center;
}

.slogan {
  color: white;
  text-shadow: 3px 3px #000;
}

.app-name{
  color: white;
  font-size: 75px;
  text-shadow: 4px 4px #000;
}

.middle{
  padding: 100px 10px;
  background-color: #ffffff;
  background-image: url('../../images/Landing2.jpg');
  background-size:100% 100%;
  background-repeat: no-repeat;
  background-position:center;
  color: white;
  text-shadow:
    0.05em 0 black,
    0 0.05em black,
    -0.05em 0 black,
    0 -0.05em black,
    -0.05em -0.05em black,
    -0.05em 0.05em black,
    0.05em -0.05em black,
    0.05em 0.05em black;
  font-size: 25px;
}

section {
  min-height: 200px;
  padding: 30px 15px;
  background-color: white;
}

section:nth-child(even) {
  background-color: rgb(244, 244, 244);
}

p {
  margin: 0 auto;
  max-width: 650px;
}

p + p {
  margin-top: 15px;
}
