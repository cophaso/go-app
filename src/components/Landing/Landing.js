import React, { Component } from 'react'
import './Landing.css';
import SignUpForm from '../../components/SignUpForm/SignUpForm'

export default class LandingPage extends Component {
  static defaultProps = {
    history: {
      push: () => {},
    },
  }

  handleRegistrationSuccess = user => {
    const { history } = this.props
    history.push('/login')
  }

  render() {
    return (
      <div>
          <header role="banner" className="header-landing">
            <h1 className="app-name">Go!</h1>
            <h2 className="slogan">Adventure is a path.</h2>
          </header>
          <section>
            <header>
                <h3>Travel without regret.</h3>
            </header>
            <p>Go! helps you plan your next adventure, so there's no more fear of missing out.</p>
          </section>
          <section className='middle'>
            <header>
                <h3>See the world, your way.</h3>
            </header>
            <p>Never lose another minute of your vacation planning what to do next.</p>
          </section>
          <section>
            <header>
                <h3>Keep track of your travels</h3>
            </header>
            <p>Keep detailed plans of your journey so you can remember every moment</p>
          </section>
          <section>
            <header>
                <h3>Plan Your Journey Now</h3>
            </header>
            <SignUpForm
              onRegistrationSuccess={this.handleRegistrationSuccess}
            />
          </section>
  </div>
    )
  }
}