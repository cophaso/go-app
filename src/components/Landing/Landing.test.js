import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Landing from './Landing';

describe(`Landing component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders a Landing by default', () => {
    const wrapper = shallow(<Landing />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the Landing given props', () => {
    const wrapper = shallow(<Landing {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
