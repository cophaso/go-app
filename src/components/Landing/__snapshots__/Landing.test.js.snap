// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component renders a Landing by default 1`] = `
<div>
  <header
    className="header-landing"
    role="banner"
  >
    <h1
      className="app-name"
    >
      Go!
    </h1>
    <h2
      className="slogan"
    >
      Adventure is a path.
    </h2>
  </header>
  <section>
    <header>
      <h3>
        Travel without regret.
      </h3>
    </header>
    <p>
      Go! helps you plan your next adventure, so there's no more fear of missing out.
    </p>
  </section>
  <section
    className="middle"
  >
    <header>
      <h3>
        See the world, your way.
      </h3>
    </header>
    <p>
      Never lose another minute of your vacation planning what to do next.
    </p>
  </section>
  <section>
    <header>
      <h3>
        Keep track of your travels
      </h3>
    </header>
    <p>
      Keep detailed plans of your journey so you can remember every moment
    </p>
  </section>
  <section>
    <header>
      <h3>
        Plan Your Journey Now
      </h3>
    </header>
    <p>
      DEMO INFO 
      <br />
      Test User:   
      <br />
      username: tuser 
      <br />
      password: Password1! 
      <br />
      Directions: Login with user and make an itinerary item to start planning your trip. Add activities to that itinerary.
    </p>
    <SignUpForm
      onRegistrationSuccess={[Function]}
    />
  </section>
</div>
`;

exports[`Landing component renders the Landing given props 1`] = `
<div>
  <header
    className="header-landing"
    role="banner"
  >
    <h1
      className="app-name"
    >
      Go!
    </h1>
    <h2
      className="slogan"
    >
      Adventure is a path.
    </h2>
  </header>
  <section>
    <header>
      <h3>
        Travel without regret.
      </h3>
    </header>
    <p>
      Go! helps you plan your next adventure, so there's no more fear of missing out.
    </p>
  </section>
  <section
    className="middle"
  >
    <header>
      <h3>
        See the world, your way.
      </h3>
    </header>
    <p>
      Never lose another minute of your vacation planning what to do next.
    </p>
  </section>
  <section>
    <header>
      <h3>
        Keep track of your travels
      </h3>
    </header>
    <p>
      Keep detailed plans of your journey so you can remember every moment
    </p>
  </section>
  <section>
    <header>
      <h3>
        Plan Your Journey Now
      </h3>
    </header>
    <p>
      DEMO INFO 
      <br />
      Test User:   
      <br />
      username: tuser 
      <br />
      password: Password1! 
      <br />
      Directions: Login with user and make an itinerary item to start planning your trip. Add activities to that itinerary.
    </p>
    <SignUpForm
      onRegistrationSuccess={[Function]}
    />
  </section>
</div>
`;
